### DISCUSSION TITLE
S3 Backup Management:

### DISCUSSION THREAD

**Discussion by WolfoTheWolf** [pelican-dev/panel#38] [Ideas]
**S3 Backup Management:** 
- Full front-end version of Wings backup system, able to do the following: 
  - Specify max server size: "Define the maximum size of the server in MB. If the server exceeds this size, it will be excluded from the backup".
  - Retention days: "Define the number of days to keep the backup".
  - Specify max backups being made at one time.
  - Specify nodes that are getting backed up.
  - Specify an S3 server and be able to create multiple S3 servers that can be assigned to nodes and can be run on a backup schedule individually using cron jobs, with S3 servers being able to manage URL lifespans, style endpoints, accelerated endpoints. There is a current addon for Ptero that allows this, but I would prefer this be a built-in feature as it's already utilized to some extent by Wings.

**Comment by ItsNeil17** [comment]
good idea, there isn't a point storing backups ON the same machine as the server files. it's more just a "store this" just to restore to that point. 

if something were to happen to that wings machine, the "backup" would also be deleted

**Comment by WolfoTheWolf** [comment]
Yeah, wings does have a built in S3 backup, but its lacking some of the more comforts, will post images of what im currently using as a visual example

**Comment by WolfoTheWolf** [comment]
![image](https://github.com/pelican-dev/panel/assets/56661068/5b5ac8af-d63f-44e2-8dbc-b491bce0ca08)
![image](https://github.com/pelican-dev/panel/assets/56661068/9796790a-a979-42bc-8524-f34274cdcc1c)
![image](https://github.com/pelican-dev/panel/assets/56661068/14c5dcb1-078f-49f1-ab9d-122e0c954575)
![image](https://github.com/pelican-dev/panel/assets/56661068/ae36ded3-0aeb-401f-ad27-687189e20762)
![image](https://github.com/pelican-dev/panel/assets/56661068/6fb9e786-35bd-49c3-8c4b-9879d24ce073)

This makes it much more manageable. I am able to create multiple S3 buckets, assign S3 systems to individuals nodes, I can allow end users to be able to view the s3 backup, just incase they want a copy. Very handy, and I can create node wide backups, I loose a node, I just use the UI to transfer my S3 backup right over onto the new node.
